src/LinkDotNet.NCronJob/Registry/JobRegistry.cs

```diff
@@ -1,81 +1,21 @@
 using System.Collections.Immutable;
-using Microsoft.Extensions.Logging;
 
 namespace LinkDotNet.NCronJob;
 
-internal sealed partial class JobRegistry
+internal sealed class JobRegistry
 {
-    private readonly JobExecutor jobExecutor;
-    private readonly TimeProvider timeProvider;
-    private readonly ILogger<JobRegistry> logger;
     private readonly ImmutableArray<JobDefinition> cronJobs;
+    private readonly ImmutableArray<Type> allJobTypes;
 
-    public JobRegistry(
-        IEnumerable<JobDefinition> jobs,
-        JobExecutor jobExecutor,
-        TimeProvider timeProvider,
-        ILogger<JobRegistry> logger)
+    public JobRegistry(IEnumerable<JobDefinition> jobs)
     {
-        this.jobExecutor = jobExecutor;
-        this.timeProvider = timeProvider;
-        this.logger = logger;
-        cronJobs = [..jobs.Where(c => c.CronExpression is not null)];
+        var jobDefinitions = jobs as JobDefinition[] ?? jobs.ToArray();
+        allJobTypes = [..jobDefinitions.Select(j => j.Type)];
+        cronJobs = [.. jobDefinitions.Where(c => c.CronExpression is not null)];
     }
 
     public IReadOnlyCollection<JobDefinition> GetAllCronJobs() => cronJobs;
 
-    /// <inheritdoc />
-    public void RunInstantJob<TJob>(object? parameter = null, CancellationToken token = default)
-        where TJob : IJob => RunScheduledJob<TJob>(TimeSpan.Zero, parameter, token);
-
-    /// <inheritdoc />
-    public void RunScheduledJob<TJob>(TimeSpan delay, object? parameter = null, CancellationToken token = default)
-    {
-        token.Register(() => LogCancellationRequested(parameter));
-
-        var run = new JobDefinition(typeof(TJob), parameter, null, null);
-
-        _ = Task.Run<Task>(async () =>
```

Comment:
Is there no use-case where you would want to trigger a job regardless of the concurrency settings for that job type?

Reply:
That is a good enhancement, to be honest.
Right now, it will always go via the queue -> executor, which makes a lot of things easier (and easier to reason about).

I will add this as its own enhancement.